target_postgres/connector.py

```diff
@@ -42,6 +42,11 @@ class PostgresConnector(SQLConnector):
     allow_merge_upsert: bool = True  # Whether MERGE UPSERT is supported.
     allow_temp_tables: bool = True  # Whether temp tables are supported.
 
+    # Whether to interpret schema contentEncoding to set the column type
+    # it is an opt-in feature because it might result in data loss if the
+    # actual data does not match the schema's advertised encoding
+    interpret_content_encoding: bool = False
```

Comment:
What do you think of this approach

```suggestion
    @cached_property
    def interpret_content_encoding(self) -> bool:
        return config.get("interpret_content_encoding", False):
```

?

You might need to add an `from functools import cached_property`.